### DISCUSSION TITLE
Looking for a context example across multiple files

### DISCUSSION THREAD

**Discussion by jtwite** [xaviergonz/mobx-keystone#394] [Q&A] [answered]
The context looks really exciting but I'm not sure on the right way to approach using it across multiple files.  In the few examples I can find it appears to be created in the same file as the models using it.  React's context has a setup where you use `createContext` and `useContext` separately.  I couldn't find anything similar to `useContext` so I ended up exporting the instance returned by `createContext`.  Is that the expected implementation?

**Comment by xaviergonz** [comment, marked as answer]
yes, that's a perfectly valid usage

**Reply by jtwite** [reply]
Thank you for clarifying!  How will that work in a nextjs environment?  Is the context actually a singleton?  My biggest concern would be memory management or one users' session impacting another.

**Reply by xaviergonz** [reply]
I'm not sure I follow...

**Reply by jtwite** [reply]
My biggest concern would be whether or not the context essentially exists in the global scope.  Which could lead to it being shared between sessions on the nextjs server when the data should be unique to that request.

**Reply by xaviergonz** [reply]
I'm afraid I don't know enough about next js to be able to answer that
